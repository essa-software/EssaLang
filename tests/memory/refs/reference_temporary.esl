struct Test {
    field: u32;
    func add(this): Test {
        print("field={}!\n", this.field);
        return Test { field: this.field + 1 };
    }
    func new(f: u32): Test {
        return Test { field: f };
    }
}

func main() {
    let _ = Test::new(1111).add().add();
    let _ = Test::new(2222).add().add();
}
