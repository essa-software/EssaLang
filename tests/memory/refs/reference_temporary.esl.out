field=1111!
field=1112!
field=2222!
field=2223!
